accounts/views.py

```diff
@@ -1,3 +1,36 @@
+
+from email.mime import audio
 from django.shortcuts import render
+from django.urls import reverse_lazy
+from django.views.generic import TemplateView,CreateView
+from accounts.models import User
+from .forms import SignupForm
+from django.contrib.auth import authenticate,login
+
+
+class SignupView(CreateView):
+    model = User
+    form_class = SignupForm
+    template_name = 'accounts/signup.html'
+    success_url = reverse_lazy('home')
+
+    def form_valid(self, form):
+        response=super().form_valid(form)
+        
+        username=form.cleaned_data.get('username')
+        email=form.cleaned_data.get('email')
+        raw_pass=form.cleaned_data.get('password1')
+        User=authenticate(username=username,email=email,password=raw_pass)
+        login(self.request,User)
+        return response

```

Comment:
```suggestion
    def form_valid(self, form):
        username = form.cleaned_data.get('username')
        email = form.cleaned_data.get('email')
        password = form.cleaned_data.get('password1')
        user=authenticate(username=username, email=email, password=password)
        if user is not None
            login(self.request, user)
            return super().form_valid(form)

```
関数内ではあまり空行を挟まない方がいいと思います。また、=はスペース開けていきましょう。ただし、キーワード引数などの=にはスペース開けなくていいです。これはpythonの推奨記述のpep8に従っています。
他の=のところも基本的にはスペースを入れてください。それと、キーワード引数のところも、" , " だけでなく、" , " の後にスペースを入れてください。
UserはUser = get_user_model()にて使用しているので、userとしましょう。
また、 if user is not Noneも追加してください。